pkg/project/project.go

```diff
@@ -215,6 +219,27 @@ func generateProjectName() string {
 	return strings.Join(str, " ")
 }
 
+// findFile finds for specific file until maxRecursiveIteration is reached.
+func findFile(fp, dir, match string) (string, bool) {
```

Comment:
We could make input parameters names a more meaningful here and provide some more info in the comment. Some suggestion:
```suggestion
// findFile searches for a file with name `fileName`. Search starts in `startDir`
// and will traverse through all parent directories until the file is found,
// root directory is reached or `maxRecursiveIteration` is exceeded. File is
// used here in the sense of "everything is a file" and can also be a directory.
//
// If `fileDir` is specified, `fileName` is expected to be inside of a directory.
// E.g. `findFile("/some/path", ".git", "config")` will look for the git config
// file (`./.git/config`).
func findFile(startDir, fileDir, fileName string) (string, bool) {
```

Reply:
I'd rename it to `findFileOrDirectory` and get rid of multiple comments. Anyways I accepted your suggestion and added it.

Reply:
`findFileOrDirectory` also sounds good :+1: With this name, we can actually shorten the comment. My main intention with the comments is to describe the following behavior:
- Usage of `fileDir` param
- The fact that `fileDir` is optional and can be left empty
- That the function will traverse through parent directories until break out condition is reached.
- Break out conditions
This is a lot, because this function does a lot. In go the function signature (function name + parameter names) should be meaningful and already give you information about what it does. Additionally a comment can be used to describe more details. This is very helpful, as you do not need to read the code of the function to understand it.